api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/GenericCohortQueries.java

```diff
@@ -160,13 +160,10 @@ public CohortDefinition getPatientsBasedOnPatientStates(int program, int state)
             + " FROM patient p"
             + " INNER JOIN patient_program pg ON p.patient_id=pg.patient_id"
             + " INNER JOIN patient_state ps ON pg.patient_program_id=ps.patient_program_id "
-            + " WHERE pg.voided=0 AND ps.voided=0 AND p.voided=0 AND"
-            + " pg.program_id="
-            + program
-            + " AND ps.state="
-            + state
+            + " WHERE pg.voided=0 AND ps.voided=0 AND p.voided=0 "
+            + " AND pg.program_id=%s AND ps.state=%s AND ps.end_date is null "
```

Comment:
@ningosi @nthfloor this is the fix on transferred out we added `ps.end_date is null`